### PR TITLE
Clickup - update to ESM and use pipedream axios

### PR DESCRIPTION
**Pull request by michelle0927**

### CODE REVIEW

**Comment by dylburger** [comment]
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/clickup/actions/create-list/create-list.mjs [hunk 1]
```diff
@@ -1,11 +1,11 @@
-const common = require("../common.js");
+import common from "../common.mjs";
 
-module.exports = {
+export default {
   ...common,
   key: "clickup-create-list",
   name: "Create List",
   description: "Creates a new list",
```
Comment (jcortes) [inline components/clickup/actions/create-list/create-list.mjs:7]: Optional:

```suggestion
  description: "Creates a new list. See the docs [here](https://clickup.com/api) in either the **Lists / Create List** section or the **Spaces / Create Space** section.",
```

File: components/clickup/actions/create-subtask/create-subtask.mjs [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
-} = require("../create-task/create-task.js");
+import clickup from "../../clickup.app.mjs";
+import createTask from "../create-task/create-task.mjs";
 
-module.exports = {
+export default {
+  ...createTask,
   key: "clickup-create-subtask",
   name: "Create Subtask",
   description: "Creates a new subtask",
```
Comment (jcortes) [inline components/clickup/actions/create-subtask/create-subtask.mjs:8]: Optional:

```suggestion
  description: "Creates a new subtask. See the docs [here](https://clickup.com/api) in **Tasks / Create Task** section.",
```

File: components/clickup/actions/create-task/create-task.mjs [hunk 3]
```diff
@@ -1,11 +1,11 @@
-const common = require("../common.js");
+import common from "../common.mjs";
 
-module.exports = {
+export default {
   ...common,
   key: "clickup-create-task",
   name: "Create Task",
   description: "Creates a new task",
```
Comment (jcortes) [inline components/clickup/actions/create-task/create-task.mjs:7]: Optional:

```suggestion
  description: "Creates a new task. See the docs [here](https://clickup.com/api) in **Tasks / Create Task** section.",
```

**Review by jcortes — APPROVED** [review]
It looks just good to me. I've just added a few minor optional suggestions. In addition to that I would also recommend  updating the `version` and `main` properties as well as adding the `gitHead` property in  the `package.json` file.

**Comment by ctrlaltdylan** [comment]
@michelle0927 @dannyroosevelt are you aware of this bug - https://github.com/PipedreamHQ/pipedream/issues/2180

Does this update affect the async options to bring in the `workspace` correctly after auth?

**Comment by michelle0927** [comment]
> @michelle0927 @dannyroosevelt are you aware of this bug - #2180
> 
> Does this update affect the async options to bring in the `workspace` correctly after auth?

@ctrlaltdylan Looks like a duplicate of https://github.com/PipedreamHQ/pipedream/issues/2043. I haven't been able to reproduce it and @dannyroosevelt said he wasn't seeing the error anymore. Are you able to reproduce it?

**Comment by michelle0927** [comment]
> Sorry @michelle0927 no I have not. I just received a message via email from the user.
> 
> I shared all of the information she did in the issue I linked above.
> 
> Is there anything I can ask her to help debug further?

I'm not sure. @dylburger Any ideas about how would be best to troubleshoot this?

**Comment by dannyroosevelt** [comment]
> > Sorry @michelle0927 no I have not. I just received a message via email from the user.
> > I shared all of the information she did in the issue I linked above.
> > Is there anything I can ask her to help debug further?
> 
> I'm not sure. @dylburger Any ideas about how would be best to troubleshoot this?

Yea I cannot repro this (just tried again), but I tested this the other day -- @michelle0927 go ahead and merge when you get a chance, then @ctrlaltdylan maybe you can have the user update to the latest version of the Create Task action and give it another whirl?

**Pull request merged**